internal/helmdeployer/install.go

```diff
@@ -112,96 +127,190 @@ func (h *Helm) install(ctx context.Context, bundleID string, manifest *manifest.
 	}
 
 	if !h.useGlobalCfg {
-		mapper, err := cfg.RESTClientGetter.ToRESTMapper()
-		if err != nil {
-			return nil, err
-		}
+		mapper, _ := cfg.RESTClientGetter.ToRESTMapper()
 		pr.mapper = mapper
 	}
 
-	if install {
-		u := action.NewInstall(&cfg)
-		u.ClientOnly = h.template || (dryRunCfg.DryRun && dryRunCfg.DryRunOption == "")
-		if cfg.Capabilities != nil {
-			if cfg.Capabilities.KubeVersion.Version != "" {
-				u.KubeVersion = &cfg.Capabilities.KubeVersion
-			}
-			if cfg.Capabilities.APIVersions != nil {
-				u.APIVersions = cfg.Capabilities.APIVersions
-			}
+	return pr
+}
+
+// runInstall executes a Helm install operation with the provided configuration and values.
+// It creates an Install action, configures it, and runs the installation.
+func (h *Helm) runInstall(ctx context.Context, cfg *action.Configuration, chart *chartv2.Chart, values map[string]interface{}, releaseName, namespace string, timeout time.Duration, options fleet.BundleDeploymentOptions, pr *postRender, dryRunCfg dryRunConfig) (*releasev1.Release, error) {
```

Comment:
nit: this is a long line, it should be reformatted for easier readability, with one parameter per line.

(same remark about `runUpgrade`)